py/jpy/src/main/c/jpy_jtype.c

```diff
@@ -861,7 +891,9 @@ int JType_ConvertPythonToJavaObject(JNIEnv* jenv, JPy_JType* type, PyObject* pyA
         return JPy_AsJString(jenv, pyArg, objectRef);
     } else if (PyBool_Check(pyArg) && (type == JPy_JObject || ((*jenv)->IsAssignableFrom(jenv, JPy_Boolean_JClass, type->classRef)))) {
         return JType_CreateJavaBooleanObject(jenv, type, pyArg, objectRef);
-    } else if (JPy_IS_CLONG(pyArg) && (type == JPy_JObject || ((*jenv)->IsAssignableFrom(jenv, JPy_Integer_JClass, type->classRef)))) {
+    } else if (JPy_IS_CLONG(pyArg) && (type == JPy_JObject || (*jenv)->IsAssignableFrom(jenv, JPy_Number_JClass, type->classRef))) {
+        return JType_CreateJavaNumberFromPythonInt(jenv, type, pyArg, objectRef);
+    } else if (JPy_IS_CLONG(pyArg) && ((*jenv)->IsAssignableFrom(jenv, JPy_Integer_JClass, type->classRef))) {
```

Comment:
This code path changes a few things:

`org.jpy.PyObject#getObjectValue`, and anything that calls into jpy and says they want a return type of `Object` or `Number`, `org.jpy.PyObject#call(java.lang.Class<T>, java.lang.String, java.lang.Class<?>[], java.lang.Object[])`, ie `pyObject.call(Number.class, "my_method")`, etc.

My guess is that there aren't any callers currently that ask for `Number`, but we do have conversion code that expects an `Object` back. Namely, how we bring python script variable changes into scope for java. I think we are really trying to solve for the long vs int issue; but there may be some existing script code that could break, if it starts getting a byte or a short back. It may be reasonable to say that we always want to convert to java int, or java long; not into java byte or java short.

As currently coded,
```python
foo = 1
```
will create a java `byte` type.